### DISCUSSION TITLE
Record Deletion in Milvus v2.4.6: Deleted Records Still Visible

### DISCUSSION THREAD

**Discussion by ducanh997** [milvus-io/milvus#38366] [Q&A and General discussion] [answered]
I am using Milvus v2.4.6, deployed on a Kubernetes cluster via Helm Chart. The setup includes:
- Milvus: Running on the K8s cluster.
- Etcd, Kafka, and S3: Deployed on a physical server (outside the cluster).

I have noticed an issue where some records that are deleted remain visible indefinitely. Even after ensuring the deletion operation is completed, querying the collection still returns the deleted records.

Questions:
- What could cause deleted records to remain visible in Milvus?
- Are there any specific configurations or operational steps I need to check for Milvus, Etcd, Kafka, or S3?

**Comment by yhmo** [comment, marked as answer]
Possible reasons:
1. The consistency_level is not "Strong".  Delete requests are processed asynchrony, if you call search() immediately after delete(), and the consistency_level is not "Strong", the deleted items might be visible to search.

2. Bugs in old versions. We have fixed known issue about delete in recently versions, such as:
- v2.4.8 Rectified a problem where delete messages failed to be forwarded due to incorrect data scope handling. (https://github.com/milvus-io/milvus/pull/35313)

Try upgrading your milvus to the latest version of v2.4

**Reply by ducanh997** [reply]
Thank you. I will try upgrading Milvus. Since the deleted item remains visible indefinitely, so I don't think this issue is related to consistency.
